pkgs/stdenv/generic/make-derivation.nix

```
@@ -330,6 +330,18 @@ else let
         ++ optional (elem "host"   configurePlatforms) "--host=${stdenv.hostPlatform.config}"
         ++ optional (elem "target" configurePlatforms) "--target=${stdenv.targetPlatform.config}";
 
+      cmakeFlags = let inherit (lib) optionals; in
+        (/**/ if lib.isString cmakeFlags then lib.warn "String 'cmakeFlags' is deprecated and will be removed in release 23.05. Please use a list of strings. Derivation name: ${derivationArg.name}, file: ${pos.file or "unknown file"}" [cmakeFlags]
+         else if cmakeFlags == null      then lib.warn "Null 'cmakeFlags' is deprecated and will be removed in release 23.05. Please use a empty list instead '[]'. Derivation name: ${derivationArg.name}, file: ${pos.file or "unknown file"}" []
+         else                                 cmakeFlags)
+        ++ optionals (stdenv.hostPlatform != stdenv.buildPlatform) ([ "-DCMAKE_SYSTEM_NAME=${lib.findFirst lib.isString "Generic" (lib.optional (!stdenv.hostPlatform.isRedox) stdenv.hostPlatform.uname.system)}"]
+        ++ optionals (stdenv.hostPlatform.uname.processor != null)  [ "-DCMAKE_SYSTEM_PROCESSOR=${stdenv.hostPlatform.uname.processor}" ]
```

These inner `optionals` seem like they should be indented.

Maybe pull the entire `cmakeFlags` attribute value through `nixpkgs-fmt`? Now seems like the right time, because we're "losing" git blame anyway for this chunk.

```suggestion
            ++ optionals (stdenv.hostPlatform.uname.processor != null)  [ "-DCMAKE_SYSTEM_PROCESSOR=${stdenv.hostPlatform.uname.processor}" ]
```

---

Done